test/scheduler/tainttoleration_integration_test.go

```diff
@@ -0,0 +1,702 @@
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										envLabel: "prod",
+									},
+									MatchExpressions: []metav1.LabelSelectorRequirement{
+										{
+											Key:      regionLabel,
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{"east", "west"},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			// Create CRP with tolerations for clusters 1,2.
+			createPickAllCRPWithPolicySnapshot(crpName, affinity, policySnapshotName, buildTolerations(tolerateClusters))
+		})
+
+		It("should add scheduler cleanup finalizer to the CRP", func() {
+			finalizerAddedActual := crpSchedulerFinalizerAddedActual(crpName)
+			Eventually(finalizerAddedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to add scheduler cleanup finalizer to CRP")
+		})
+
+		It("should create scheduled bindings for clusters with tolerated taints", func() {
+			scheduledBindingsCreatedActual := scheduledBindingsCreatedOrUpdatedForClustersActual(selectedClusters, zeroScoreByCluster, crpName, policySnapshotName)
+			Eventually(scheduledBindingsCreatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to create the expected set of bindings")
+			Consistently(scheduledBindingsCreatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Failed to create the expected set of bindings")
+		})
+
+		It("should not create any binding for clusters with untolerated taints", func() {
+			noBindingsCreatedActual := noBindingsCreatedForClustersActual(unSelectedClusters, crpName)
+			Eventually(noBindingsCreatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Some bindings have been created unexpectedly")
+			Consistently(noBindingsCreatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Some bindings have been created unexpectedly")
+		})
+
+		It("should report status correctly", func() {
+			statusUpdatedActual := pickAllPolicySnapshotStatusUpdatedActual(selectedClusters, unSelectedClusters, policySnapshotName)
+			Eventually(statusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update status")
+			Consistently(statusUpdatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Failed to update status")
+		})
+
+		AfterAll(func() {
+			// Remove taints
+			removeTaintsFromMemberClusters(taintClusters)
+			// Delete the CRP.
+			ensureCRPAndAllRelatedResourcesDeletion(crpName)
+		})
+	})
+
+	// This is a serial test as adding taints, tolerations can affect other tests.
+	Context("pick N clusters with affinity specified, ignore valid clusters with taints, CRP has some matching tolerations after update", Serial, Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		policySnapshotName1 := fmt.Sprintf(policySnapshotNameTemplate, crpName, 1)
```

Comment:
```suggestion
		currentPolicySnapshotName := fmt.Sprintf(policySnapshotNameTemplate, crpName, 1)
```

or oldPolicySnapshotName